app/apollo/resolvers/serviceSubscription.js

```diff
@@ -96,6 +96,16 @@ const serviceResolvers = {
         });
       }
 
+      // render cluster information if users ask for
+      if(queryFields.cluster && serviceSubscriptions) {
+        const clusterIds = _.map(serviceSubscriptions, 'clusterId');
+        const clusters = await models.Cluster.getClustersByIds(clusterIds);
+        serviceSubscriptions = serviceSubscriptions.map((sub)=>{
+          sub.cluster = clusters[sub.clusterId];
+          return sub;
+        });
+      }
+
```

Comment:
This is best off inside of the applyQueryFieldsToSubscriptions() function
The applyQueryFields functions call eachother when necessary, which lets you do deeper nesting of requested attributes. So if a user was accessing a different resolver which returned serviceSubscriptions, they can also ask for cluster info inside of those, and it will know to run this block of code to include them.

Reply:
Done